tests/components/progettihwsw/test_config_flow.py

```diff
@@ -0,0 +1,168 @@
+    assert result["step_id"] == "user"
+    assert result["errors"] == {}
+
+    mock_value_step_rm = {
+        "relay_1": "bistable",  # Mocking a single relay board instance.
+    }
+
+    with patch(
+        "homeassistant.components.progettihwsw.config_flow.ProgettiHWSWAPI.check_board",
+        return_value=mock_value_step_user,
+    ):
+        result2 = await hass.config_entries.flow.async_configure(
+            result["flow_id"],
+            {CONF_HOST: "", CONF_PORT: 80},
+        )
+
+    assert result2["type"] == RESULT_TYPE_FORM
+    assert result2["step_id"] == "relay_modes"
+    assert result2["errors"] == {}
+
+    with patch(
+        "homeassistant.components.progettihwsw.async_setup",
+        return_value=True,
+    ), patch(
+        "homeassistant.components.progettihwsw.async_setup_entry",
+        return_value=True,
+    ):
+        result3 = await hass.config_entries.flow.async_configure(
+            result2["flow_id"],
+            mock_value_step_rm,
+        )
+
+    assert result3["type"] == RESULT_TYPE_CREATE_ENTRY
+    assert result3["data"]
+    assert result3["data"]["title"] == "1R & 1IN Board"
+    assert result3["data"]["is_old"] is False
+    assert result3["data"]["relay_count"] == result3["data"]["input_count"] == 1
+
+
+async def test_form_wrong_info(hass):
+    """Test we handle wrong info exception."""
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": config_entries.SOURCE_USER}
+    )
+
+    assert result["step_id"] == "user"
+
+    with patch(
+        "homeassistant.components.progettihwsw.config_flow.ProgettiHWSWAPI.check_board",
+        return_value=mock_value_step_user,
+    ):
+        result2 = await hass.config_entries.flow.async_configure(
+            result["flow_id"], {CONF_HOST: "", CONF_PORT: 80}
+        )
+
+    assert result2["type"] == RESULT_TYPE_FORM
+    assert result2["step_id"] == "relay_modes"
+    assert result2["errors"] == {}
+
+    result3 = await hass.config_entries.flow.async_configure(
+        result2["flow_id"], {"relay_1": ""}
+    )
+
+    assert result3["type"] == RESULT_TYPE_FORM
+    assert result3["step_id"] == "relay_modes"
+    assert result3["errors"] == {"base": "wrong_info_relay_modes"}
+
+
+async def test_form_cannot_connect(hass):
+    """Test we handle unexisting board."""
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": config_entries.SOURCE_USER}
+    )
+
+    assert result["step_id"] == "user"
+
+    with patch(
+        "homeassistant.components.progettihwsw.config_flow.ProgettiHWSWAPI.check_board",
+        return_value=False,
+    ):
+        result2 = await hass.config_entries.flow.async_configure(
+            result["flow_id"],
+            {CONF_HOST: "", CONF_PORT: 80},
+        )
+
+    assert result2["type"] == RESULT_TYPE_FORM
+    assert result2["step_id"] == "user"
+    assert result2["errors"] == {"base": "cannot_connect"}
```

Comment:
Codacy found an issue: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=6090606)